Update feat/dev-medtrum branch to bring in updates from MedtrumKit

marionbarker opened the pull request:
## Purpose

Update MedtrumKit.

Tagging @bastiaanv 

## Method

The last version of MedtrumKit included in Trio branch feat/dev-medtrum is from 09-Aug-2025.

This PR brings the most recent commit from MedtrumKit into Trio
* Caveat - I am preparing the PR for the MedtrumKit experts to approve
* I tested it using a Medtrum pump on a test phone

This update advances commits from MedtrumKit: 5bfb806...e695e62, which comprises 30 commits.

## Test


Test phone is SE 2nd gen running iOS 18.6.2.
* CGM is from Nightscout
* Pump is Medtrum
   * Built on phone: feat/dev-medtrum commit aa34d2976
   * Pump start was nominal
   * Note - I did this test prior to pushing this commit to Trio
* Built `updates_to_medtrumkit`, commit bef8d7f3a on same phone
   * Closed loop continued and manual boluses were accepted
   * Temp Basal shows on Trio main screen and Medtrum screen
   * Several SMB were reported and I could hear them on the pump

MikePlante1 committed:
Liquid Glass – Hard Pass

marionbarker committed:
update MedtrumKit: 5bfb806...e695e62

marionbarker committed:
reformat swiftformat exclude line and add MedtrumKit

marionbarker commented:
## Comments

There are 2 linting questions:

1. In MedtrumKit, two files had white space modified by Xcode when building.

My first attempt to fix this was to add MedtrumKit to the items excluded from the `Trio/scripts/swiftformat.sh` script.
* I reformatted the `--exclude` line to make it easier to read
* I added both DanaKit and MedtrumKit to the `--exclude` line
* I believe that change is required for Trio so pushed this commit even though it did not affect the linting

2. When building, there is an update to `Trio/Resources/InfoPlist.xcstrings` that I think is correct. But I did not want to commit it without checking.

```
diff --git a/Trio/Resources/InfoPlist.xcstrings b/Trio/Resources/InfoPlist.xcstrings
index e8388f2cc..82e1dead5 100644
--- a/Trio/Resources/InfoPlist.xcstrings
+++ b/Trio/Resources/InfoPlist.xcstrings
@@ -8,7 +8,7 @@
         "en" : {
           "stringUnit" : {
             "state" : "new",
-            "value" : "Trio Medtrum"
+            "value" : "Trio"
           }
         }
       }
@@ -20,7 +20,7 @@
         "en" : {
           "stringUnit" : {
             "state" : "new",
-            "value" : "Trio Medtrum"
+            "value" : "Trio"
           }
         }
       }
```

marionbarker commented:
## Comment

I had to delete the Medtrum pump twice to get Trio to return to the Add Pump option under Insulin Pump.

bastiaanv commented:
LGTM.

`When building, there is an update to Trio/Resources/InfoPlist.xcstrings that I think is correct. But I did not want to commit it without checking.`

Regarding this, that is my mistake. I have 2 trio apps on my phone, the Trio app I use to test Medtrum is called: `Trio Medtrum`. The git diff looks good and should be applied to this PR!

The double pump deletion needs to be checked. Hope to have some time for that tonight

marionbarker committed:
restore app name to Trio, fix inadventent commit

Sjoerd-Bo3 committed:
Merge pull request #765 from MikePlante1/smash-the-glass

Disable Liquid Glass

github-actions[bot] committed:
CI: Bump APP_DEV_VERSION to 0.5.1.25 [skip ci]

bastiaanv commented:
@marionbarker Could you update the SHA of MedtrumKit to [94fc976](https://github.com/loopandlearn/MedtrumKit/commit/94fc976fe525dbaee13470f2e73e68089d373bfc)?

Changelog:
- chore: change default AlarmType to BeepOnly instead of Silence
- fix: detect pump base change

Are there open topics on this PR @marionbarker ?

marionbarker committed:
Merge branch 'dev' into updates_to_medtrumkit

marionbarker committed:
update MedtrumKit: default alert now beep; detect pump base change, translations

marionbarker commented:
## Update

Pulled in the latest updates for MedtrumKit.
Pulled in latest Trio dev branch updates.

Successful test with caveat about silent vs beep alerts mentioned below.

## Test

Test with branch `updates_to_medtrumkit`, commit a26e87a87

Build on iOS 26 SE 3rd gen phone using Xcode 16.4
* successful build
* the alarm setting does not appear to have changed, still defaults to silent
* attached brain and started priming

> So it works but not quite as advertised. I suspect that is because the first time I attached a Medtrum, the alert state was set to silent. That was probably kept in the app.

The code review clearly shows that `alarmSetting` should be `.BeepOnly` if not already set.

marionbarker commented:
## Test continued

This is about the 5th time I have used this patch. I got a fault after priming. The pump was not beeping until I modified the setting, yet again, to Beep from Silence.

I tested the delete of the pump and it stayed gone this time.

I interleaved by adding an Omnipod DASH, then removed that and went to Medtrum Nano.

It is still showing up with default Alarm setting of silence, even though I changed it to Beep before - several times.

marionbarker commented:
> Are there open topics on this PR?

For this PR - 
 - I don't think the beep vs silence is fixed yet
 - However, this PR is an improvement over current state and I support merging as is.

For a later PR -
- There is no way to cancel from the intro screen except to pull down the card - which does work, but does not leave an add pump icon on the main screen. 
- If instead you go to the 4th(?) screen and select delete pump, then the add pump icons shows on the main screen. 

Not a MedtrumKit only problem and for a different PR
- even after you see the add pump icon on the main screen, if you quit and restart the app, the add pump icon has gone away, leaving the old pump icon. 
- When you touch that, you are taken to the add pump screen.

bastiaanv commented:
I changed the MedtrumKit UIController to do the following:
- Every page, during onboarding, has either a "Go back" or "Cancel"-button on the top left
- After a pump base serial number has been confirmed, the pumpManager is considered "onboarded", which should remove the "add pump" button
- Delete pump is only shown if pumpManager is "onboarded"
- (also fixed the BeepOnly in the Onboarding flow)

New SHA: [8b46b60](https://github.com/loopandlearn/MedtrumKit/commit/8b46b60835072f02340de351d99b7c1ca2264ec1)

I also noticed that the expiration icon is missing after onboarding. This might be a simulator issue, will check it later
<img width="518" height="406" alt="Screenshot 2025-09-17 at 14 12 19" src="https://github.com/user-attachments/assets/94a2a3dd-8b12-4151-b340-e7ba2050d7ba" />

marionbarker committed:
update MedtrumKit: fix onboarding logic PR 24
